src/data/datainterfaces.ts

```diff
@@ -1,9 +1,76 @@
 import { IInstanceEnvironment } from "../environment";
-import { IInstanceDetails, IRiskAssessmentValue, State } from "../instance/instanceinterfaces";
+import { IInstanceDetails, IRiskAssessmentValue, IRiskAssessmentValueSchema, State, StateSchema } from "../instance/instanceinterfaces";
 import { UserDetails } from "../user";
 import { Process, ActionHandler } from "..";
-import { RiskAssessmentCycle } from "../riskassessment/riskassessmentinterfaces";
+import { RiskAssessmentCycle, RiskAssessmentCycleSchema } from "../riskassessment/riskassessmentinterfaces";
 import { IProcessDetails } from "../process";
+import Joi from "joi";
+
+export type FieldType =
+  | "ProcessHubTextInput"
+  | "ProcessHubDateTime"
+  | "ProcessHubTextArea"
+  | "ProcessHubInstanceTitle"
+  | "ProcessHubFileUpload"
+  | "ProcessHubRoleOwner"
+  | "ProcessHubDate"
+  | "ProcessHubDropdown"
+  | "ProcessHubChecklist"
+  | "ProcessHubRadioButton"
+  | "ProcessHubDecision"
+  | "ProcessHubRoxFile"
+  | "ProcessHubSignature"
+  | "ProcessHubLabel"
+  | "ProcessHubMail"
+  | "ProcessHubNumber"
+  | "ProcessHubRiskAssessment"
+  | "ProcessHubRiskAssessmentTodos"
+  | "ProcessHubRiskAssessmentCycle"
+  | "ProcessHubSpreadSheet"
+  | "ProcessHubProcessLink"
+  | "ProcessHubInstanceNumber"
+  | "ProcessHubRoxFileLink"
+  | "ProcessHubCalculatedField";
+
+const FieldTypeSchema = Joi.string().pattern(
+  new RegExp(
+    "^ProcessHubTextInput$|" +
+      "^ProcessHubDateTime$|" +
+      "^ProcessHubTextArea$|" +
+      "^ProcessHubInstanceTitle$|" +
+      "^ProcessHubFileUpload$|" +
+      "^ProcessHubRoleOwner$|" +
+      "^ProcessHubDate$|" +
+      "^ProcessHubDropdown$|" +
+      "^ProcessHubChecklist$|" +
+      "^ProcessHubRadioButton$|" +
+      "^ProcessHubDecision$|" +
+      "^ProcessHubRoxFile$|" +
+      "^ProcessHubSignature$|" +
+      "^ProcessHubLabel$|" +
+      "^ProcessHubMail$|" +
+      "^ProcessHubNumber$|" +
+      "^ProcessHubRiskAssessment$|" +
+      "^ProcessHubRiskAssessmentTodos$|" +
+      "^ProcessHubRiskAssessmentCycle$|" +
+      "^ProcessHubSpreadSheet$|" +
+      "^ProcessHubProcessLink$|" +
+      "^ProcessHubInstanceNumber$|" +
+      "^ProcessHubRoxFileLink$|" +
+      "^ProcessHubCalculatedField$|",
+  ),
+);
```

Comment:
I would prefer to also have an enum test similar to the other one I proposed - just to check all values of the original enum to make sure no one forgot one that is available in the typesdef.